FreeRTOS_TCP_IP.c

```diff
@@ -3725,12 +3725,22 @@
         FreeRTOS_Socket_t * pxReturn = NULL;
         uint32_t ulInitialSequenceNumber;
 
-        /* Assume that a new Initial Sequence Number will be required. Request
-         * it now in order to fail out if necessary. */
-        ulInitialSequenceNumber = ulApplicationGetNextSequenceNumber( *ipLOCAL_IP_ADDRESS_POINTER,
-                                                                      pxSocket->usLocalPort,
-                                                                      pxTCPPacket->xIPHeader.ulSourceIPAddress,
-                                                                      pxTCPPacket->xTCPHeader.usSourcePort );
+        /* Silently discard a SYN packet which was sent to the broadcast address
+         * 255.255.255.255. */
+        if( pxTCPPacket->xIPHeader.ulDestinationIPAddress != 0xFFFFFFFF )
```

Comment:
Aniruddha, I think that we should also test for a local broadcast address "x.x.x.255", and for multicast addresses. Maybe in a separate function:
~~~c
BaseType_t xIsUnicastAddress( uint32_t ulIPAddress )
{
    BaseType_t xResult = pdTRUE;

    if( ( FreeRTOS_ntohl( ulIPAddress ) & 0xffU ) == 0xffU )
    {
        /* This is a broadcast address x.x.x.255. */
        xResult = pdFALSE;
    }
    else if( xIsIPv4Multicast( ulIPAddress ) != pdFALSE )
    {
        /* This is a multicast address. */
        xResult = pdFALSE;
    }

    return xResult;
}
~~~
And so make this change:
~~~c
-    if( pxTCPPacket->xIPHeader.ulDestinationIPAddress != 0xFFFFFFFF )
+    if( xIsUnicastAddress( pxTCPPacket->xIPHeader.ulDestinationIPAddress ) != pdFALSE )
~~~

Reply:
I agree, added.
Thank you.